### PR TITLE
Fix crash when running tuist build with verbose config variable

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/3684

### Short description 📝

We were executing [business logic](https://github.com/tuist/tuist/compare/fix-crash-environment?expand=1#diff-c10d893a0d13a5433625b2ea899cb54210648ccd49e3de2301be7245ad471abdL77) when initializing `Environment` which could lead to accessing `Environment` while the `Environment` was being initalized (leading to a crash as described in the [attached issue](https://github.com/tuist/tuist/issues/3684)).

To remedy this, I have added `bootstrap` method that will be executed in `main.swift` to execute the business logic only _after_ `Environment.shared` is initialized. 

cc @kikeenrique

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

**Review by adellibovi — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 
 ### Fixed
 
+- Crash when running `tuist build` with `TUIST_CONFIG_VERBOSE=1` [#3752](https://github.com/tuist/tuist/pull/3752) by [@fortmarek](https://github.com/fortmarek)
```
Comment (danyf90) [inline CHANGELOG.md:9]: [i] I prefer to write `Fix`, even if it looks like a repetition :)
```suggestion
- Fix crash when running `tuist build` with `TUIST_CONFIG_VERBOSE=1` [#3752](https://github.com/tuist/tuist/pull/3752) by [@fortmarek](https://github.com/fortmarek)
```

**Review by danyf90 — APPROVED** [review]

**Review by pepicrft — APPROVED** [review]

**Review by luispadron — APPROVED** [review]
Just ran into this thanks for the fix 🙏🏼

Re-ran the jobs because we got a test failure

File: projects/tuist/features/step_definitions/shared/tuist.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@
   # On CI we expect tuist to be built already by the previous job `release_build`, so we skip `swift build`
 
   if ENV["CI"].nil?
-    ["tuist", "ProjectDescription", "ProjectAutomation"].each do |product|
+    ["tuist", "ProjectDescription", "ProjectAutomation", "tuistenv"].each do |product|
```
Comment (luispadron) [inline projects/tuist/features/step_definitions/shared/tuist.rb:9]: This is causing test failures for some reason 🤔

**Review by luispadron — APPROVED** [review]

**Review by pepicrft — APPROVED** [review]

**Comment by pepicrft** [comment]
Pushed a fix here. If CI is happy I'll go ahead and merge. Thanks @fortmarek for taking care of this.

**Pull request merged**